common/src/test/java/uk/gov/justice/services/common/converter/ListToJsonArrayConverterTest.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.justice.services.common.converter;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.isA;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.doThrow;
+
+import java.util.Arrays;
+import java.util.UUID;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import uk.gov.justice.services.common.converter.exception.ConverterException;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ListToJsonArrayConverterTest {
+	private static final UUID ID_ONE = UUID.randomUUID();
+	private static final UUID ID_TWO = UUID.randomUUID();
+	private static final String NAME_ONE = "POJOONE";
+	private static final String NAME_TWO = "POJOTWO";
+
+	@Rule
+	public ExpectedException exception = ExpectedException.none();
+
+	@Mock
+	private ObjectMapper mapperMock;
+
+	@Test
+	public void shouldConvertListToJsonArray() {
+		ListToJsonArrayConverter<Pojo> listToJsonArraysConverter = new ListToJsonArrayConverter<Pojo>();
+		listToJsonArraysConverter.mapper = new JacksonMapperProducer().objectMapper();
+		listToJsonArraysConverter.stringToJsonObjectConverter = new StringToJsonObjectConverter();
+
+		Pojo pojoOne = new Pojo(ID_ONE, NAME_ONE);
+		Pojo pojoTwo = new Pojo(ID_TWO, NAME_TWO);
+		JsonArray jsonArray = listToJsonArraysConverter.convert(Arrays.asList(pojoOne, pojoTwo));
+		assertThat(jsonArray, equalTo(expectedJsonArray()));
+
+	}
+
+	@Test
+	public void shouldThrowExceptionOnConversionError() {
+		ListToJsonArrayConverter<Pojo> listToJsonArraysConverter = new ListToJsonArrayConverter<Pojo>();
+		exception.expect(ConverterException.class);
+		listToJsonArraysConverter.convert(null);
+
+	}
+
+	@Test
+	public void shouldThrowIllegalArgumentException() throws JsonProcessingException {
+		ListToJsonArrayConverter<Pojo> listToJsonArraysConverter = new ListToJsonArrayConverter<Pojo>();
+		listToJsonArraysConverter.mapper = mapperMock;
+		listToJsonArraysConverter.stringToJsonObjectConverter = new StringToJsonObjectConverter();
+
+		Pojo pojoOne = new Pojo(null, null);
+		doThrow(JsonProcessingException.class).when(mapperMock).writeValueAsString(pojoOne);
```

Comment:
Shouldn't we throw IOException here, because that's the top level exception thrown by writeValueAsString and that's what we are catching in our code?